### DISCUSSION TITLE
Issue with counting stock for certain parts in InvenTree

### DISCUSSION THREAD

**Discussion by palagap** [inventree/InvenTree#5407] [Q&A]
Hi there, 

We noticed an counting parts issue that affects some of the parts we have loaded into InvenTree, but not others. For example, when we try to count the stock of this part, it looks like the following:

<img width="1681" alt="Screenshot 2023-08-04 at 10 23 37 AM" src="https://github.com/inventree/InvenTree/assets/40808444/583cdff5-238f-4437-b524-65743f61cea2">

Clicking on the “Count part stock” option results in this blank display:

<img width="1244" alt="Screenshot 2023-08-03 at 4 46 49 PM" src="https://github.com/inventree/InvenTree/assets/40808444/e8c02e18-6e85-4fd5-92bd-5373da3416c8">

On the other hand, for another part, we see the following:

<img width="1669" alt="Screenshot 2023-08-03 at 4 47 22 PM" src="https://github.com/inventree/InvenTree/assets/40808444/6df43f4e-85bf-46e5-adce-687fc4fde9b8">

Clicking on the “Count part stock” option now results in the following, as expected:

<img width="1245" alt="Screenshot 2023-08-03 at 4 47 44 PM" src="https://github.com/inventree/InvenTree/assets/40808444/ac6056b1-cdba-42a8-b4b8-13211b3d9677">

Most parts seem to have no issue with the count part stock option, but a few do. We compared part settings between the parts that don’t have this issue and those that do, and could not find any differences. Do you have any suggestions on how to resolve this problem? We are running the latest stable version of InvenTree.

The InvenTree demo seems to also have the same issue for some parts:

<img width="1686" alt="Screenshot 2023-08-07 at 2 02 08 PM" src="https://github.com/inventree/InvenTree/assets/40808444/4b9ec120-2f46-4c1d-8a8d-d3965a510384">

Clicking on “Count part stock”:

<img width="1247" alt="Screenshot 2023-08-07 at 2 02 17 PM" src="https://github.com/inventree/InvenTree/assets/40808444/8689e6a6-4be0-451f-9bca-c275774b252d">

**Comment by SchrodingersGat** [comment]
@palagap you might see this message if all the stock item objects you have selected are "serialized" - these can't be "counted". Is this the case maybe?

**Reply by palagap** [reply]
Unfortunately that's not the case - we verified that the parts that are presenting a problem do not have serial numbers. Some of the parts in the InvenTree demo that have this issue are:

https://demo.inventree.org/part/77/#
https://demo.inventree.org/part/81/#
https://demo.inventree.org/part/88/#

**Reply by SchrodingersGat** [reply]
I had a look at the parts you listed, and all the available stock is actually serialized (see below):

![image](https://github.com/inventree/InvenTree/assets/10080325/d3b44621-6698-446e-8e70-c0b1a4585bcf)

There should certainly be a better information message for the user here, but this lines up with my comments above - serialized stock items cannot be "counted"

**Reply by palagap** [reply]
Yes, you're right - the ones from the demo are serialized, but the ones in our system are not and still have the issue. Here's what it looks like from our end:

![Screenshot 2023-08-09 at 10 18 26 AM](https://github.com/inventree/InvenTree/assets/40808444/6a2d87b7-2c90-402f-ad10-2fa57fea27b7)

**Reply by SchrodingersGat** [reply]
Can you reproduce this with any parts on the demo server which have non-serialized stock items?

**Reply by palagap** [reply]
Unfortunately we can't replicate this on the demo server; it just seems to be an issue on our end. We're going to try duplicating the part and making the old one inactive; hopefully that fixes the issue.
